alpha-lab/imu-transformations/index.md

```diff
@@ -202,93 +196,17 @@ def spherical_to_cartesian_scene(elevations, azimuths):
     ).T
 ```
 
-Now we have the tools to convert 3D gaze data to world coordinates:
+Now we can transform the data to world coordinates. Since we are dealing with 3D directions, rather than 3D points here, it does not make sense to apply the translation that we used in the `transform_scene_to_world` function above. We are thus setting it to zero here.
 
 ```python
 def gaze_3d_to_world(gaze_elevation, gaze_azimuth, imu_quaternions):
     cart_gazes_in_scene = spherical_to_cartesian_scene(gaze_elevation, gaze_azimuth)
-    return transform_scene_to_world(cart_gazes_in_scene, imu_quaternions)
-```
-
-## 2D Gaze to World Coordinates
-
-If you are starting from [the 2D gaze values in scene image coordinates (i.e., `gaze x/y [px]`)](https://docs.pupil-labs.com/neon/data-collection/data-format/#gaze-csv), then you will need to first [undistort](https://docs.pupil-labs.com/alpha-lab/undistort/) and unproject those points to obtain the corresponding 3D gaze vectors. Note that this requires [loading the scene camera calibration data](https://docs-staging.pupil-labs.com/alpha-lab/undistort/#reading-from-the-cloud-download-json-file).
-
-```python
-def unproject_2d_gaze(gaze_points_2d, scene_camera_matrix, scene_distortion_coefficients):
-    """
-    Transform the 2D gaze values from Neon to 3D gaze vectors.
-    """
-
-    gaze_points_2d_undist = cv2.undistortPoints(gaze_points_2d, scene_camera_matrix, scene_distortion_coefficients)
-    gaze_vectors_3d = cv2.convertPointsToHomogeneous(gaze_points_2d_undist)
-    gaze_vectors_3d /= np.linalg.norm(gaze_vectors_3d, axis=2)[:, np.newaxis]
-    
-    return gaze_vectors_3d
-```
-
-Then, we can use the functions from [the previous section](#3d-gaze-to-world-coordinates) to convert 2D gaze to 3D world coordinates:
-
-```python
-# The gaze data and the IMU quaternions should be sampled at the
-# same timestamps. You can linearly interpolate the IMU data to
-# ensure this.
-# See the Application Example:
-# http://docs.pupil-labs.com/alpha-lab/imu-transformations/#application-example
-
-gaze_points_2d = gaze[["gaze x [px]", "gaze y [px]"]].to_numpy()
-def gaze_2d_to_world(gaze_points_2d, scene_camera_matrix, scene_distortion_coefficients, imu_quaternions):
-    cart_gazes_in_scene = undistort_and_unproject(
-        gaze_points_2d, scene_camera_matrix, scene_distortion_coefficients
-    )
-    return transform_scene_to_world(
-        cart_gazes_in_scene, imu_quaternions
-    )
-```
-
-## World Spherical Coordinates
```

Comment:
This too is not related to IMU transformations. I'm not sure we'd want to cover the transformation between cartesian and spherical coordinates anywhere explicitly in our docs, as this is an algebra problem that's covered a lot on the internet and it is not required that often in eye tracking. Happy to discuss this though!

Reply:
@marc-tonsen In our case, I say we need it, because Neon's spherical coordinate conventions break with traditional spherical coordinate conventions in subtle ways. Considering the IMU seems to be a frequent source of confusion for users, I would vote to make things as easy as possible for them. It's also been a request from two users so far.